lib/util/http/host-rules.ts

```diff
@@ -68,46 +68,59 @@ function findMatchingRules(options: GotOptions, url: string): HostRule {
 }
 
 // Apply host rules to requests
-export function applyHostRules(url: string, inOptions: GotOptions): GotOptions {
-  const options: GotOptions = { ...inOptions };
+export function applyHostRules(
+  url: string,
+  inOptions: HttpOptions
+): HttpOptions['gotOptions'] {
```

Comment:
```suggestion
): GotOptions {
```

Use interface directly